src/corvus_python/README.md

```diff
@@ -4,3 +4,86 @@ This package is an interim home for data-related & python-based utility function
+
+### Recommendations when running in Synapse
+
+These recommendations are intended for the scenario when you are deploying packages into a Spark cluster to be executed from a Synapse notebook, although there will be equivalent implementations for other platforms.
+
+#### Add the Azure Monitor configuration to the notebook which calls your code
+
+Adding the configuration to the notebook rather than inside your package means that it will be possible to apply different Open Telemetry configuration elsewhere - for example, in test runs - or leave it out completely if required to effectively disable tracing.
+
+#### Service name and service instance Id
+
+Use your package name as the service name. This will be accesible in App Insights as the `Role name` in every span that is sent. If applying a filter in the transaction search screen, it is referred to as `Cloud role name`.
+
+Use a suitable correlation Id for the service instance Id. This will be accessible in App Insights as the `Role instance` in every span that is sent. If applying a filter in the transaction search screen, it is referred to as `Cloud role instance`.
+
+When code is executed from a Synapse notebook, it's likely that it will be being invoked from a pipeline. In this case you can set `service.instance.id` to the Pipeline run Id to provide an easy link between pipeline runs in the Synapse monitor. This can either be passed into the notebook as a parameter, or accessed in the notebook using `mssparkutils.runtime.context["pipelinejobid"]`.
+
+#### Span types
+
+The supplied decorators create spans of kind `SpanKind.INTERNAL` - the default. This is intended to indicate that the span represents an internal operation within an application, as opposed to an operations with remote parents or children. This will result in a trace of type `DEPENDENCY` in App Insights (also referred to as `InProc`).
+
+Consider wrapping calls to your package with a manually created span of kind `SpanKind.REQUEST`. This will result in a trace of type `REQUEST` in App Insights, and will make it easier to locate invocations of your package by searching for spans of type `REQUEST` using your correlation Id.
+
+If your package has a small number of entrypoints, you can create the Request span inside your package. Alternatively, you can create the Request span in the notebook which uses your package. The code looks like this:
+
+```python
+from opentelemetry import trace
+
+tracer = trace.get_tracer(__name__)
+
+def my_entrypoint():
+
+    with tracer.start_as_current_span(
+            "my_entrypoint_name",
+            kind=trace.SpanKind.SERVER) as span:
```

Comment:
Above you refer to `SpanKind.REQUEST` but the example shows `SpanKind.SERVER` - which is correct?